apis/Google.Cloud.Metadata.V1/Google.Cloud.Metadata.V1/MetadataClientImpl.cs

```diff
@@ -0,0 +1,702 @@
+        /// </para>
+        /// </remarks>
+        /// <exception cref="TaskCanceledException">
+        /// The timeout in the <see cref="HttpClient"/> elapses before the response comes back from the server.
+        /// </exception>
+        /// <returns>The changed value(s) for an endpoint or a JSON object with the changed contents of the directory.</returns>
+        /// <seealso cref="InstanceMetadataChanged"/>
+        /// <seealso cref="ProjectMetadataChanged"/>
+        public override string WaitForChange(string key) =>
+            GetResult(() => WaitForChangeAsync(key));
+
+        /// <summary>
+        /// Waits for changes to the value or values specified by the relative URL asynchronously.
+        /// </summary>
+        /// <param name="key">The metadata key on which to wait for changes, such as "instance/scheduling/automatic-restart"</param>
+        /// <remarks>
+        /// <para>
+        /// If the key specified is a metadata endpoint, the result will be the value, possibly separated by newlines if there are multiple values.
+        /// If the key specified is a directory, a recursive request must be made and the result will be a JSON object containing all values in the directory.
+        /// For example, to wait for changes on all instance metadata use a key of "instance?recursive=true".
+        /// </para>
+        /// <para>
+        /// By default, if half the length of the timeout in the <see cref="HttpClient"/> elapses before any changes occur, this will return the current value.
+        /// To override the timeout, add a query string after the key with a timeout_sec field. For example, "instance/tags?timeout_sec=60". Note that if the
+        /// the timeout in the <see cref="HttpClient"/> elapses before the server's timeout elapses, as specified by timeout_sec, an exception will occur.
+        /// </para>
+        /// <para>
+        /// See https://cloud.google.com/compute/docs/storing-retrieving-metadata#project_and_instance_metadata for more information on available keys.
+        /// </para>
+        /// <para>
+        /// To detect any changes at the project or instance level, use a <paramref name="key"/> of "project" or "instance".
+        /// Alternatively, use <seealso cref="ProjectMetadataChanged"/> or <seealso cref="InstanceMetadataChanged"/>, respectively.
+        /// </para>
+        /// </remarks>
+        /// <exception cref="TaskCanceledException">
+        /// The timeout in the <see cref="HttpClient"/> elapses before the response comes back from the server.
+        /// </exception>
+        /// <returns>A task containing the changed value(s) for an endpoint or a JSON object with the changed contents of the directory.</returns>
+        /// <seealso cref="InstanceMetadataChanged"/>
+        /// <seealso cref="ProjectMetadataChanged"/>
+        public override async Task<string> WaitForChangeAsync(string key, CancellationToken cancellationToken = default(CancellationToken))
```

Comment:
There's no validation here, which makes me somewhat nervous. Do we know enough about the useful values that we *could* validate?